src/commonMain/kotlin/baaahs/sim/FakeNetwork.kt

```diff
@@ -24,6 +24,10 @@ class FakeNetwork(
     private val httpServersByPort:
             MutableMap<Pair<Network.Address, Int>, FakeLink.FakeHttpServer> = hashMapOf()
 
+    private var services = mutableMapOf<String, FakeLink.FakeMdns.FakeMdnsService>()
```

Comment:
I think these are always `FakeRegisteredService`s, yeah? Which lets you lose some casts below.

Reply:
This is maybe I don’t understand exactly what FakeNetwork is for, is it for the Simulator? I think it’s for the simulator. The point is, the brains and pinky need to be able to communicate through the FakeNetwork (hence the shared storage at Network-level instead of Link- or Mdns-level), but also brains shouldn’t be able to update or unregister Pinky's services, nor vice versa. That’s where most of the conditional as? casts come from -- ideally, the component that registered the service should be the only one for whom that succeeds.

Obviously this isn’t written that way though. Do you have any suggestions on how to make it do what I want?

The TestNetwork has the same issue.

Reply:
Ok, I think I fixed it just by moving the `unregister` implementation to the Fake/TestRegisteredService